ckan_cloud_operator/providers/solr/solrcloud/manager.py

```diff
@@ -94,13 +94,13 @@ def initialize(interactive=False, dry_run=False):
     _config_set('sc-main-host-name', solrcloud_host_name)
     logs.info(f'Initialized solrcloud service: {solrcloud_host_name}')
 
-    # TODO - need to check the pod names and ensure these are solrcloud ones
-    # TODO - actual number is _+ 1_ and not _+ 2_
-    expected_running = len(sc_host_names) + len(zk_host_names) + 2
+    expected_running = len(sc_host_names) + len(zk_host_names)
     while True:
         pods = kubectl.get('pods')
+        sc_app_name = _get_resource_labels(for_deployment=True, suffix='sc')['app']
+        zk_app_name = _get_resource_labels(for_deployment=True, suffix='zk')['app']
         running = len([x for x in pods['items']
-                       if x['status']['phase'] == 'Running'])
+                       if x['status']['phase'] == 'Running' and x['metadata']['labels']['app'] in [zk_app_name, sc_app_name]])
```

Comment:
X['metadata']['labels']['app'].startswith('provider-solr-solrcloud')
But replace the string literal with the proper constants

Reply:
done